codec-http/src/test/java/io/netty/handler/codec/http/multipart/HttpPostRequestEncoderTest.java

```diff
@@ -75,6 +77,56 @@ public void testSingleFileUpload() throws Exception {
     }
 
     @Test
+    public void testAllowedMethods() throws Exception {
+        shouldThrowExceptionIfNotAllowed(HttpMethod.CONNECT);
+        shouldThrowExceptionIfNotAllowed(HttpMethod.PUT);
+        shouldThrowExceptionIfNotAllowed(HttpMethod.POST);
+        shouldThrowExceptionIfNotAllowed(HttpMethod.PATCH);
+        shouldThrowExceptionIfNotAllowed(HttpMethod.DELETE);
+        shouldThrowExceptionIfNotAllowed(HttpMethod.GET);
+        shouldThrowExceptionIfNotAllowed(HttpMethod.HEAD);
+        shouldThrowExceptionIfNotAllowed(HttpMethod.OPTIONS);
+        try {
+            shouldThrowExceptionIfNotAllowed(HttpMethod.TRACE);
+            fail("Should raised an exception with TRACE method");
+        } catch (ErrorDataEncoderException e) {
+            // Exception is willing
+        }
+    }
+
+    private void shouldThrowExceptionIfNotAllowed(HttpMethod method) throws Exception {
+        DefaultFullHttpRequest request = new DefaultFullHttpRequest(HttpVersion.HTTP_1_1,
+                                                                    method, "http://localhost");
+
+        HttpPostRequestEncoder encoder = new HttpPostRequestEncoder(request, true);
+        File file1 = new File(getClass().getResource("/file-01.txt").toURI());
+        encoder.addBodyAttribute("foo", "bar");
+        encoder.addBodyFileUpload("quux", file1, "text/plain", false);
+
+        String multipartDataBoundary = encoder.multipartDataBoundary;
+        String content = getRequestBody(encoder);
+
+        String expected = "--" + multipartDataBoundary + "\r\n" +
```

Comment:
can we share this code with `testSingleFileUpload`? Seems like there is some duplication here.